### PR TITLE
Add Docker support for mviewer

### PR DESCRIPTION
**Pull request by RemiDesgrange**
Simple Dockerfile to be able to launch mviewer quickly. 

It does not take care of `config.xml`. You have to create it yourself, or load it via a volume.

### CODE REVIEW

**Comment by Gaetanbrl** [comment]
How it works with customLayer / customControls and templates ?

**Comment by Gaetanbrl** [comment]
It's a pleasure to get more contributions but please, create an issue first before a PR.

https://mviewerdoc.readthedocs.io/fr/latest/doc_contrib/contrib.html#proposer-une-modification

**Comment by RemiDesgrange** [comment]
> How it works with customLayers / customControls and templates ?

You create a volume ? Or override the docker image. Maybe a better solution is needed here...

> It's a pleasure to get more contributions but please, create an issue first before a PR.

This PR is "part" of the mviewer studio docker PR. But I'll remember it ;-)

**Comment by Gaetanbrl** [comment]
> Maybe a better solution is needed here...

About Docker ?

**Comment by RemiDesgrange** [comment]
hum no, maybe we should add a build arg or something. I'll think about it.

**Comment by bchartier** [comment]
I think it's useful to know how to deploy mviewer in a docker container.
But mviewer is just html/css/javascript.
In order to deploy it (with Docker or not) you don't need much effort.
Deploying it for test purpose with Docker you just need something like this (if you run this command at the root directory of mviewer):
`docker run -dit --name mviewer -p 8084:80 -v "$PWD":/usr/local/apache2/htdocs/ httpd:2.4`
(I use a simple Apache container image but it should by the same with other web servers)

You mentioned this PR as part of the Flask mviewerstudio backend PR. In that PR the docker-compose does not need a Dockerfile for mviewer.

That's why I think the simplest way to introduce docker in this project would be to mention this in the README.md. Adding a Dockerfile does not seem useful to me.

**Comment by fvanderbiest** [comment]
> Adding a Dockerfile does not seem useful to me.

I do not agree.
There are cases where mounting a volume with the source code is not an option.

In addition, having it does not harm.
So, I'm +1 for adding a dockerfile with a gh action building it and pushing the image to docker hub.

**Comment by bchartier** [comment]
> > Adding a Dockerfile does not seem useful to me.
> 
> I do not agree.

Do not pay attention to my initial comment ; I was probably drunk.

**Comment by fvanderbiest** [comment]
> It does not take care of `config.xml`. You have to create it yourself, or load it via a volume.

Have to find another solution...
I'd like mviewer to use `config.xml` from a subdirectory, so we can mount a volume onto the directory (so it can be runtime-edited). Mounting a file is not always an option with orchestrators.

@RemiDesgrange this does not seem complex, see https://github.com/geobretagne/mviewer/blob/a6d9e0713837611d4c52e0abeddab67310fda124/index.html#L130-L132

The file could also be renamed into `config/default.xml`

File: README.md [hunk 1]
```diff
@@ -25,6 +25,20 @@ Le déploiement se passe en trois étapes :
     Vous avez maintenant un visualiseur géographique fonctionnel avec les couches de la Région Bretagne
     3. Si vous souhaitez publier vos propres couches/thèmes, modifiez le fichier config.xml
 
+## Docker
+
+Si vous souhaitez faire tourner mviewer dans un conteneur docker, un `Dockerfile` est a votre disposition.
+
+```bash
+# batir l'image docker
+docker build -t mviewer .
```
Comment (fvanderbiest) [inline README.md]: geobretagne/mviewer

**Comment by fvanderbiest** [comment]
> How it works with customLayers / customControls and templates ?

I guess you'll have to mount volumes onto the corresponding folders.
Can you send us a complete use case @Gaetanbrl ?

File: README.md (hunk 1 above)
Comment (fvanderbiest) [reply README.md]: Github action required here too.

**Comment by pierrejego** [comment]
@fvanderbiest yes one volumes should be mount for apps folder. You can see more information here https://mviewerdoc.readthedocs.io/fr/latest/doc_dev/config_practices.html#organisation-des-autres-fichiers
In the same time, we should be able to export configuration file generated by mviewerstudio to this same volume

I hope this is clear enough, do not hesitate to call me for to discuss about this point

**Comment by spelhate** [comment]
@RemiDesgrange can you target the develop branch instead of master for this PR ? Thanks.

**Comment by fvanderbiest** [comment]
@pmauduit will take care of it

File: README.md (hunk 1 above)
Comment (pmauduit) [reply README.md]: actually mviewer/mviewer (see similar pr on mviewerstudio)

**Comment by pmauduit** [comment]
> In the same time, we should be able to export configuration file generated by mviewerstudio to this same volume

@pierrejego shall we consider creating a config/ subdirectory, or do you advise reusing `apps/` instead ?

**Comment by bchartier** [comment]
Is this pull request ready for review?
I see that @fvanderbiest and @pmauduit are identified as reviewers.
Do you need I review from somebody else?

**Comment by fvanderbiest** [comment]
Superseded by #236
Should be closed. Thanks

**Pull request closed without merging**